tests/test_fixtures.py

```diff
@@ -1046,3 +1046,42 @@ def func(handler_map={}):
         return wrapper
     return decorator
 
+
+class NonDaemonPool(pool.Pool):
+    def Process(self, *args, **kwds):
+        proc = super(NonDaemonPool, self).Process(*args, **kwds)
+
+        class NonDaemonProcess(proc.__class__):
+            """Monkey-patch process to ensure it is never daemonized"""
+
+            @property
+            def daemon(self):
+                return False
+
+            @daemon.setter
+            def daemon(self, val):
+                pass
+
+        proc.__class__ = NonDaemonProcess
+
+        return proc
+
+
+class TestRunner:
+    def __init__(self, tests):
+        self._tests = tests
+        self._test_runner_pool = NonDaemonPool(min(len(self._tests), cpu_count())   )
+
+    @staticmethod
+    def test_runner(test):
+        logger.info('>>> Running {}...'.format(test.__name__))
+        test_run_result = test()
+        logger.info('>>> {}: {}\n'.format('OK' if test_run_result else 'FAILED', test.__name__))
+        return test_run_result
+
+    def run(self):
+        results = self._test_runner_pool.map(TestRunner.test_runner, self._tests)
+        total_result = True
+        for result in results:
+            total_result = total_result and result
+        return total_result
```

Comment:
Does the output from the tests running simultaneously get interleaved?

Reply:
Not exactly, total_result will be False if at least one of the tests fails, it's kind of result for the overall test suite.

Reply:
Sorry, I meant that I was wondering if the output printed to the screen from the tests gets mixed together from multiple tests running in parallel or if ctest or whatever is smart enough to separate them somehow.

Reply:
@patrickvacek did you actually clarify that or is the question still valid?

Reply:
Still an open question, actually.